plugins/woocommerce/client/blocks/assets/js/blocks/checkout/test/block.js

```diff
@@ -157,6 +164,98 @@ describe( 'Testing Checkout', () => {
 		expect( fetchMock ).toHaveBeenCalledTimes( 1 );
 	} );
 
+	it( 'Allows saving payment method if the customer is creating an account or has already logged in', async () => {
+		await act( async () => {
+			const PaymentMethodContent = () => <div>A payment method</div>;
+			registerPaymentMethod( {
+				name: 'test-payment-method',
+				label: 'Payment method with cards',
+				content: <PaymentMethodContent />,
+				edit: <PaymentMethodContent />,
+				icons: null,
+				canMakePayment: () => true,
+				supports: {
+					showSavedCards: true,
+					showSaveOption: true,
+					features: [ 'products' ],
+				},
+				ariaLabel: 'Test Payment Method',
+			} );
+		} );
+
+		const { rerender } = render( <CheckoutBlock /> );
+
+		await waitFor( () => expect( fetchMock ).toHaveBeenCalled() );
+		expect(
+			screen.getByText( /Payment method with cards/i )
+		).toBeInTheDocument();
```

Comment:
```suggestion
		expect(
			await screen.findByText( /Payment method with cards/i )
		).toBeInTheDocument();
```
This way, you avoid checking the implementation details of the component and wait for what the customer expects. Not feeling strongly about it though, there are cases where waiting for the fetch to finish is much simpler. Proceed as you wish 😊